src/odemis/driver/test/semnidaq_test.py

```diff
@@ -550,15 +567,36 @@ def test_downsample(self):
         self.assertEqual(data[0, 0], buffer[margin])
         self.assertEqual(data[-1, -1], buffer[-1])
 
-    def compute_expected_metadata(self) -> Tuple[Tuple[int, int], Tuple[float, float], float]:
-        exp_shape = self.scanner.resolution.value[::-1]
-        scanner_pxs = self.scanner.pixelSize.value
-        scale = self.scanner.scale.value
-        exp_pxs = tuple(p * s for p, s in zip(scanner_pxs, scale))
-        exp_duration = self.scanner.dwellTime.value * numpy.prod(exp_shape)
-        return exp_shape, exp_pxs, exp_duration
+        # Test like CI
+        res=(1024, 768)
+        margin=10
+        osr = 1
+        prev_samples_n = 0
+        prev_samples_sum = 0
+        data = numpy.zeros(res[::-1], dtype=numpy.int16)  # reset
+        grain = 100000
+        buffer = numpy.linspace(500, 1500, (res[1] * (res[0] + margin)) * osr, dtype=numpy.uint32)
+        left_buffer = buffer[:]
+        acc_dtype = numpy.uint32
+        acquired_n = 0
+        while left_buffer.size > 0:
+            logging.debug(f"Passing another {grain} samples, still {left_buffer.size} left")
+            prev_samples_n, prev_samples_sum = Acquirer._downsample_data(data, res, margin,
+                                                                         acquired_n, osr,
+                                                                         left_buffer[:grain],
+                                                                         prev_samples_n, prev_samples_sum,
+                                                                         acc_dtype,
+                                                                         average=False)
+            acquired_n += grain
+            left_buffer = left_buffer[grain:]
+
+        self.assertEqual(samples_n, 0)
+        #numpy.testing.assert_array_equal(data, data_osr1)
```

Comment:
why did you comment this?

Reply:
Cannot be easily used. => just removed it .